### ISSUE TITLE
Update settings to adopt project ID instead of project path

### ISSUE THREAD

**Issue by cguedes** [refstudio/refstudio#445]
We should update this code in `projectEventListeners.ts` to use a new key.

```js
// TODO: This method needs to be refactored to have use a project id instead of a path
function persistProjectIdInSettings(id: string) {
  setCachedSetting('project.current_directory', id);
  void saveCachedSettings();
}
```

Tasks:
- [x] Add `active_project_id` to the settings backend API
- [x] Adopt the new `active_project_id` in the client code
- [x] #494
- [x] #504

No setting migration is needed.

This is a follow-up task after #425

**Comment by cguedes** [comment]
CC @gjreda

**Comment by cguedes** [comment]
@cguedes will do this change

**Comment by cguedes** [comment]
@gjreda can you create a new setting entry for `active_project_id`?
This will replace (and deprecate) the setting `current_directory` to persist the current open project across sessions.

**Comment by gjreda** [comment]
Related to this, do we want to remove `migrate_settings`? IIRC we're ok with breaking changes at this point in the project, and just needing to re-add your API key isn't a huge deal.

**Comment by cguedes** [comment]
Reopen as we are still missing some UI tasks.

**Comment by cguedes** [comment]
This is now finally done 🎉
